libr/core/cmd_anal.c

```diff
@@ -5065,16 +5073,24 @@ static void cmd_anal_syscall(RCore *core, const char *input) {
 			} else {
 				n = r_syscall_get_num (core->anal->syscall, input + 2);
 				if (n != -1) {
-					r_cons_printf ("%d\n", n);
+					if (!strcmp (os, "macos")) {
+						r_cons_printf ("0x%02x\n", n);
+					} else {
+						r_cons_printf ("%d\n", n);
+					}
 				} else {
 					eprintf ("Unknown syscall name\n");
 				}
 			}
 		} else {
 			list = r_syscall_list (core->anal->syscall);
 			r_list_foreach (list, iter, si) {
-				r_cons_printf ("%s = 0x%02x.%u\n",
+				if (!strcmp (os, "macos")) {
```

Comment:
why don't you print `si->swi` for macos as well? And why can't we just switch from `%u` to `0x%x` for all architectures?

Reply:
strcmps inside a loop is going to slowdown a lot. add an bool isMacOS = !strcmp (os, "macos"); before the rListForeach() and just check for this variable in there